disk/disk_linux.go

```diff
@@ -298,6 +298,14 @@ func PartitionsWithContext(ctx context.Context, all bool) ([]PartitionStat, erro
 				}
 			}
 
+			if strings.HasPrefix(d.Device, "/dev/mapper/") {
+				devpath, err := os.Readlink(d.Device)
+				if err != nil {
+					return nil, err
+				}
+				d.Device = "/dev/" + filepath.Base(devpath)
```

Comment:
It may be more failsafe to follow the symlink with `devpath, err := filepath.EvalSymlinks(d.Device)` and affecting the result on success with `d.Device = devpath` instead of assuming `os.Readlink` will return `../dm-X` here.

Reply:
I'm fixing it in master.